### PR TITLE
Add documentation for primary rebalancing

### LINKED ISSUE #6790: [DOC] Add Documentation for new cluster shard placement strategies
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 
https://github.com/opensearch-project/OpenSearch/pull/12656 adds new setting in the upcoming release. The issue is to add documentation around the same.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by Arpit-Bandejiya**
### Description
Add documentation for new setting introduced for Opensearch rebalancing.

### Issues Resolved
Closes #6790 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Arpit-Bandejiya** [comment]
@hdhalter Sorry for the delay. Raised the PR for the two new settings which are raised.

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 1]
```diff
@@ -64,6 +64,10 @@ OpenSearch supports the following cluster-level routing and shard allocation set
 
 - `cluster.routing.allocation.balance.prefer_primary` (Boolean): When set to `true`, OpenSearch attempts to evenly distribute the primary shards between the cluster nodes. Enabling this setting does not always guarantee an equal number of primary shards on each node, especially in the event of failover. Changing this setting to `false` after it was set to `true` does not invoke redistribution of primary shards. Default is `false`.
 
+- `cluster.routing.allocation.rebalance.primary.enable` (Boolean): When set to `true`, OpenSearch attempts to evenly rebalance the primary shards between the cluster nodes. Enabling this setting, cluster tries to maintain number of primary shards on each node with maximum buffer defined by `cluster.routing.allocation.rebalance.primary.buffer`. Changing this setting to `false` after it was set to `true` does not invoke redistribution of primary shards. Default is `false`.
```
Comment (Naarcha-AWS) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:67]:
```suggestion
- `cluster.routing.allocation.rebalance.primary.enable` (Boolean): When set to `true`, OpenSearch attempts to rebalance the primary shards between the cluster nodes. When enabled, the cluster tries to maintain number of primary shards on each node with the maximum buffer defined by the `cluster.routing.allocation.rebalance.primary.buffer` setting . Changing this setting to `false` after it was set to `true` does not invoke the redistribution of primary shards. Default is `false`.
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

**Review by Naarcha-AWS — APPROVED** [review]

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 2]
```diff
@@ -64,6 +64,10 @@ OpenSearch supports the following cluster-level routing and shard allocation set
 
 - `cluster.routing.allocation.balance.prefer_primary` (Boolean): When set to `true`, OpenSearch attempts to evenly distribute the primary shards between the cluster nodes. Enabling this setting does not always guarantee an equal number of primary shards on each node, especially in the event of failover. Changing this setting to `false` after it was set to `true` does not invoke redistribution of primary shards. Default is `false`.
 
+- `cluster.routing.allocation.rebalance.primary.enable` (Boolean): When set to `true`, OpenSearch attempts to rebalance the primary shards between the cluster nodes. When enabled, the cluster tries to maintain number of primary shards on each node with the maximum buffer defined by the `cluster.routing.allocation.rebalance.primary.buffer` setting . Changing this setting to `false` after it was set to `true` does not invoke the redistribution of primary shards. Default is `false`.
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:67]:
```suggestion
- `cluster.routing.allocation.rebalance.primary.enable` (Boolean): When set to `true`, OpenSearch attempts to rebalance the primary shards between the cluster nodes. When enabled, the cluster tries to maintain the number of primary shards on each node, with the maximum buffer defined by the `cluster.routing.allocation.rebalance.primary.buffer` setting. Changing this setting to `false` after it was set to `true` does not invoke the redistribution of primary shards. Default is `false`.
```

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS Just a few changes. Thanks!

**Pull request merged**